### PR TITLE
Cascading values and parameters UE pass

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #19286

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters?view=aspnetcore-3.1&branch=pr-en-us-21366)

* Going to fix an API cross-link scenario across the Blazor docs in passing here. I've been thinking about how to handle attribute links over the years. I want to go with the format (using `[JSInvokable]` as an example) ...

  > \[\`[JSInvokable]\` attribute](xref:Microsoft.JSInterop.JSInvokableAttribute)

  ... everywhere over ...

  > \[\`[JSInvokable]\`](xref:Microsoft.JSInterop.JSInvokableAttribute) attribute

  ... because the API uses the word "attribute" and because the code styling of the attribute (`[...]`) slightly obscures that it's a clickable link when no additional text is present. When "attribute" is included in the link text, it's easy to see that it's a clickable thing.

* I won't bore with a long list of all the **_BIG_** *little nit* updates here; but IMO, this UE pass turned out really, **_really_** well. Before reviewing, read the live topic at https://docs.microsoft.com/aspnet/core/blazor/components/cascading-values-and-parameters and then read the [internal review topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters?view=aspnetcore-3.1&branch=pr-en-us-21366). **_Booyeah!_** *Now, that's what I'm talking about!* :beers:

* The new *Cyberdyne Systems Snippet Builder 5000*&trade; :smile: has run amok here and is trying to **_KILL_** this PR. I'm inquiring offline how to deal with this guy ...

  terminator

  *The Terminator* &copy;1984 [Skydance Media](https://skydance.com/)

  **UPDATE**: Conferred with Cyberdyne offline, and I was able to take this guy down by updating the way that snippets are hosted for the topic. We're going to be moving all snippets into projects and compiling them every PR. I add a common sample app on this PR for that purpose that I'll be using as I go forward for snippets across the Blazor docs.

  Right before he died tho :point_up: he said ........ yeah! ....... **_"I'll be back!"_** 😄

### CODE REVIEW

File: aspnetcore/blazor/components/cascading-values-and-parameters.md [hunk 1]
```diff
@@ -110,41 +88,145 @@ To cascade multiple values of the same type within the same subtree, provide a u
...
     [Parameter]
-    public MyCascadingType ParentCascadeParameter2 { get; set; }
+    public CascadingType ParentCascadeParameter2 { get; set; }
 
     ...
 }
 ```
 
-In a descendant component, the cascaded parameters receive their values from the corresponding cascaded values in the ancestor component by name:
+In a descendant component, the parameters receive their cascaded values from the ancestor component by <xref:Microsoft.AspNetCore.Components.CascadingValue%601.Name%2A>:
```
Comment (captainsafia) [inline aspnetcore/blazor/components/cascading-values-and-parameters.md:100]:
```suggestion
In a descendant component, the cascaded parameters receive their cascaded values from the ancestor component by <xref:Microsoft.AspNetCore.Components.CascadingValue%601.Name%2A>:
```

I think having the "cascaded" descriptor makes it a little clearer.

**Review by captainsafia — COMMENTED** [review]
Posted some prelim comments inline.

**Comment by guardrex** [comment]
@captainsafia On the last commit, I went ahead and added Blazor Server samples (3.x and 5.x) for the future. I'm going to need "BlazorSample"-named Blazor Server apps, too, for cross-linking snippets under the new snippet building paradigm. In the topic, this only required a few sample app cross-link path changes. You don't need to re-review anything that you already looked at.

**Comment by guardrex** [comment]
@captainsafia It's ok if you're 🏃😅 and can't look. I need to merge this fairly soon. This is blocking further snippet work because it has the new common snippet sample apps on it.

**Review by captainsafia — APPROVED** [review]

**Comment by guardrex** [comment]
Thanks @captainsafia ... I'll look it over a last time Wednesday morning to make sure that I don't 💥 the whole Internet with this 🙈.

**Pull request merged**